scripts/shared/compile.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+  
+## Process command line flags ##
+
+source ${SCRIPTS_DIR}/lib/shflags
+DEFINE_boolean 'debug' false "Build the binary with debug information included (or stripped)"
+DEFINE_boolean 'upx' true "Use UPX to make the binary smaller"
+DEFINE_string 'ldflags' '' "Extra flags to send to the Go compiler"
+FLAGS_HELP="USAGE: $0 [--[no]debug] [--[no]upx] [--ldflags '<flags>'] binary source"
+FLAGS "$@" || exit $?
+eval set -- "${FLAGS_ARGV}"
+
+if [[ $# != 2 ]]; then
+    echo "The binary and source must be sepcified!"
+    exit 1
+fi
+
+[[ "${FLAGS_debug}" = "${FLAGS_TRUE}" ]] && build_debug=true || build_debug=false
+[[ "${FLAGS_upx}" = "${FLAGS_TRUE}" ]] && build_upx=true || build_upx=false
+ldflags=${FLAGS_ldflags}
+binary=$1
+source_file=$2
+
+set -e
+
+source ${SCRIPTS_DIR}/lib/debug_functions
+
+## Main ##
+
+mkdir -p ${binary%/*}
+
+echo "Building ${binary@Q} (ldflags: ${ldflags@Q})"
+if [ "$build_debug" = "false" ]; then
+    ldflags="-s -w ${ldflags}"
+fi
+
+CGO_ENABLED=0 go build -ldflags "${ldflags}" -o $binary $source_file
+[[ "$build_upx" = "false" ]] || upx $binary
```

Comment:
I think it would be better to skip `upx` for debug builds too.

Reply:
I was leaving it to the caller but I guess it wont hurt to have it hardcoded that way